Add retry notes to supported clients. (#291)
memgraph/documentation: pages/client-libraries/rust.mdx

- [Manual transaction management](#manual-transaction-management)
- [Implicit transaction management](#implicit-transaction-management)

> If you face conflicting transactions because of write-write conflict, you will have to retry transactions manually. It is recommended to run them as exponential backoff, with some randomization to avoid deadlocks. 

Reviewer: ```suggestion
You will have to retry transactions manually if you face conflicting transactions because of write-write conflict. It's recommended to run them as exponential backoff, with randomization to avoid deadlocks. 
```
